### ISSUE TITLE
Why is this client so awful? 

### ISSUE THREAD

**Issue by mihaiav** [kubernetes/client-go#819]
I'm trying to develop a service which manages kubernates batch jobs. Almost 1 hour passed and I still didn't manage to make it work. Why is this client so awfully designed? Looks like generated code

**Comment by liggitt** [comment]
I'm sorry you are having trouble using the client. Much of the client is generated, though that is intended to make use of the different APIs more consistent. Have you tried using some of the examples to get started?

Example running outside a cluster:
https://github.com/kubernetes/client-go/blob/master/examples/out-of-cluster-client-configuration

Example running inside a pod:
https://github.com/kubernetes/client-go/tree/master/examples/in-cluster-client-configuration

To work with batch jobs, you could replace `clientset.CoreV1().Pods` in those examples with `clientset.BatchV1().Jobs`

**Comment by mihaiav** [comment]
Thanks for the links! In the end I managed to create the "jobs" service I needed. Authenticating using a Google Service Account JSON key passed as a slice of bytes ([]byte) was a bit tricky. I had to create my own `AuthProviderPlugin`. 

After getting a bit more familiar with the package I'm not sure anymore if my initial criticism was fair. The client is definitely comprehensive and its design starts to make sense.
